internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```diff
@@ -193,7 +196,7 @@ NLBCustomDomainRole:
     ManagedPolicyArns:
       - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
 {{- end}}
-{{- if eq .NLB.Listener.Protocol "TLS" }}
+{{- if .NLB.CertificateRequired }}
```

Comment:
Out of curiosity why this change? isn't it equivalent to before.

Or is the idea that in the next PRs we will support multiple listeners and we need a single field to check across all listeners if there is `tls` as protocol?

Reply:
Yes, you are right. For now there is only one listener so we can easily pick a value from `.NLB.Listener.Protocol` but going forward we can't keep spanning all the listeners. If we have even one listener with TLS then we will set this flag to true and call cert create and validation function.